Formula/wb32-dfu-updater_cli.rb

```diff
@@ -0,0 +1,31 @@
+class Wb32DfuUpdaterCli < Formula
+  desc "USB programmer"
+  homepage "https://github.com/WestberryTech/wb32-dfu-updater"
+  url "https://github.com/WestberryTech/wb32-dfu-updater/archive/refs/tags/1.0.0.tar.gz"
+  sha256 "2b1c5b5627723067168af9740cb25c5c179634e133e4ced06028462096de5699"
+  license "Apache-2.0"
+  head "https://github.com/WestberryTech/wb32-dfu-updater.git", branch: "master"
+
+  depends_on "cmake" => :build
+  depends_on "libusb"
+
+  def install
+    args = std_cmake_args
+
+    libusb = Formula["libusb"]
+    args << "-DLIBUSB_INCLUDE_DIRS=#{libusb.opt_include}/libusb-1.0"
+    args << "-DLIBUSB_LIBRARIES=#{libusb.opt_lib/shared_library("libusb-1.0")}"
+
+    system "cmake", "-S", ".", "-B", "build", *args
+    system "cmake", "--build", "build"
+    system "cmake", "--install", "build"
+  end
+
+  test do
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.